src/Esri.ArcGISRuntime.Toolkit/Shared/UI/Legend/LayerLegend.cs

```diff
@@ -0,0 +1,185 @@
+                    return;
+                }
+            }
+
+            var items = new ObservableCollection<LegendInfo>();
+            ctrl.ItemsSource = items;
+            LoadRecursive(items, LayerContent, ShowEntireTreeHiarchy);
+        }
+
+        private async void LoadRecursive(IList<LegendInfo> itemsList, ILayerContent content, bool recursive)
+        {
+            if (content == null)
+            {
+                return;
+            }
+
+            try
+            {
+#pragma warning disable ESRI1800 // Add ConfigureAwait(false) - This is UI Dependent code and must return to UI Thread
+                var legendInfo = await content.GetLegendInfosAsync();
+#pragma warning restore ESRI1800
+                foreach (var item in legendInfo)
+                {
+                    itemsList.Add(item);
+                }
+            }
+            catch
+            {
+                return;
+            }
+
+            if (recursive)
+            {
+                foreach (var item in content.SublayerContents)
+                {
+                    LoadRecursive(itemsList, item, recursive);
+                }
+            }
+        }
+
+        private void Layer_Loaded(object sender, EventArgs e)
+        {
+            (sender as Layer).Loaded -= Layer_Loaded;
+#if NETFX_CORE
+            var ignore_ = Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, UpdateLegend);
+#else
+            var ignore = Dispatcher.InvokeAsync(UpdateLegend);
+#endif
+        }
+
+        /// <summary>
+        /// Gets or sets the ItemsTemplate
+        /// </summary>
+        public DataTemplate ItemTemplate
+        {
+            get { return (DataTemplate)GetValue(ItemTemplateProperty); }
+            set { SetValue(ItemTemplateProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="ItemTemplate"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty ItemTemplateProperty =
+            DependencyProperty.Register(nameof(ItemTemplate), typeof(DataTemplate), typeof(LayerLegend), new PropertyMetadata(null));
+
+        /// <summary>
+        /// Gets or sets the Items Panel Template
+        /// </summary>
+        public ItemsPanelTemplate ItemsPanel
+        {
+            get { return (ItemsPanelTemplate)GetValue(ItemsPanelProperty); }
+            set { SetValue(ItemsPanelProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="ItemsPanel"/> dependency property.
+        /// </summary>
+        public static readonly DependencyProperty ItemsPanelProperty =
+            DependencyProperty.Register(nameof(ItemsPanel), typeof(ItemsPanelTemplate), typeof(LayerLegend), new PropertyMetadata(null));
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the entire <see cref="ILayerContent"/> tree hiarchy should be rendered
```

Comment:
The word "Hierarchy" is misspelled. Suggested fix: https://gist.github.com/fragmer/398288c085b6233e1bb3a7ed8c026d25